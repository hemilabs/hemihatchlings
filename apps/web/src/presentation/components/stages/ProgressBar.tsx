interface ProgressBarProps {
  percent: number
}

function ProgressBar({ percent }:ProgressBarProps): JSX.Element {
  const totalBars = 18
  const activeBars = Math.round((totalBars * percent) / 100)
  const deactivateBars = totalBars - activeBars
  const bars = []
  let nextKey = 1

  for (let i = 0; i < activeBars; i++) {
    bars.push(<div key={nextKey} className='bg-water-blue w-7 h-14'></div>)
    nextKey++
  }

  for (let i = 0; i < deactivateBars; i++) {
    bars.push(<div key={nextKey} className='bg-water-blue/40 w-7 h-14'></div>)
    nextKey++
  }

  return (
    <div className="flex flex-row justify-center gap-3">
      {bars}
    </div>
  )
}

export default ProgressBar
