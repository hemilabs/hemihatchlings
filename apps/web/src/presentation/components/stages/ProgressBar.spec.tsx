import { describe, it, expect} from 'vitest'
import { RenderResult, render } from '@testing-library/react'
import ProgressBar from './ProgressBar'

const renderComponent = (): RenderResult => {
    return render(
      <ProgressBar percent={50} />
    )
  }

describe('src/presentation/ProgressBar', () => {
  it('renders correctly', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })
})
