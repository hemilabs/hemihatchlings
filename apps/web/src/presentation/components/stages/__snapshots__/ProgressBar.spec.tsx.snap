// Vitest Snapshot v1

exports[`src/presentation/ProgressBar > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row justify-center gap-3"
  >
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
    <div
      class="bg-water-blue/40 w-7 h-14"
    />
  </div>
</DocumentFragment>
`;
